File: PArticulo2/PArticulo2/Articulo.cs
using System;

namespace PArticulo2
{
	public class Articulo
	{
		public Articulo (long id, string nombre, decimal? precio, long? categoria) {
			Id = id;
			Nombre = nombre;
			Precio = precio;
			Categoria = categoria;
		}
		public long Id { get; set;}
		public string Nombre { get; set;}
		public decimal? Precio { get; set;}
		public long? Categoria { get; set;}
	}
}

File: HMySql/src/main/java/org/institutoserpis/ad/Articulo.java
package org.institutoserpis.ad;

import java.math.BigDecimal;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

@Entity
public class Articulo {
	private long Id;
	private String nombre;
	private BigDecimal precio;
	private Categoria categoria;

	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	public long getId() {
		return Id;
	}
	public void setId(long id) {
		Id = id;
	}
	public String getNombre() {
		return nombre;
	}
	public void setNombre(String nombre) {
		this.nombre = nombre;
	}
	public BigDecimal getprecio() {
		return precio;
	}
	public void setprecio(BigDecimal precio) {
		this.precio = precio;
	}
	@ManyToOne
	@JoinColumn(name="categoria")
	public Categoria getCategoria(){
		return categoria;
	}
	public void setCategoria(Categoria categoria){
		this.categoria=categoria;
	}
	@Override
	public String toString() {
		return String.format("%s %s %s (categoria=%s)", Id, nombre, precio, categoria);
	}
}
